src/main/java/roboy/context/TimestampedValueHistory.java

```diff
@@ -0,0 +1,91 @@
+package roboy.context;
+
+import java.util.Iterator;
+import java.util.NavigableSet;
+import java.util.TreeMap;
+
+/**
+ * Sample implementation of a ValueHistory using timestamps (longs) as keys
+ * and a TreeMap for data storage.
+ *
+ * The timestamps are equal or larger than the time when updateValue() was called.
+ * Implementation does not guarantee perfect timestamp accuracy, but achieves key uniqueness.
+ *
+ */
+public class TimestampedValueHistory<V> implements AbstractValueHistory<Long, V> {
+    /**
+     * Marks the last time a value was added to the history (or initialization).
+     */
+    private volatile long lastTime;
+    private TreeMap<Long, V> data;
+    private int totalValuesAdded;
+
+    public TimestampedValueHistory() {
+        data = new TreeMap<>();
+        lastTime = System.nanoTime();
```

Comment:
Just a comment:
``System.nanoTime()`` is very useful if you get a lot of values in very short intervals. However, the timestamps no not provide the "real" time so you can actually use them for logging (like "on May xy, at 2 PM something happened"). If we need timestamps with real time, we should use ``System.currentTimeMillis()``. Disadvantage is smaller precision. 

I do not really know how often values will arrive so this point is open for discussion. If the update interval is less than 20 ms, we should probably stick to ``System.nanoTime()``.